### ISSUE TITLE
Support request

### ISSUE THREAD

**Issue by rmoosa** [justinschoeman/ModbusThermostat#2]
Hi. Apologies for logging this as an issue. Wasn't sure how else to reach out to you. 
Very interested in your project. 
I'm an absolute beginner to arduino. Don't know how to read the diagram in your code. 
How's exactly does this get wired up to the arduino?
Do we need to add any other components? Or does the Geyserwise thermostat get connected directly to the arduino?

**Comment by justinschoeman** [comment]
Yeah, sorry - this is not really a beginner project. It involves desoldering the MCU from the Geyserwise controller, and replacing it with an Arduino.

I am not sure if this will work for you (it depends on which Geyserwise system you have):
https://www.thingwala.com/geyserwala/connect/

But this allows you some DIY control without in-depth hardware knowledge.

**Comment by rmoosa** [comment]
I bought one of the newer tuya integrated geyserwise units, but I'm not impressed with the restrictions. 
1 - integration to tuya platform is restricted to trial period and licencing is the price of a car. 
2 - the timeslots for temp settings are restricted and not adjustable. 

I was exploring hooking up the thermostat to a sonoff th elite and writing some software to do the controls from there.

**Comment by justinschoeman** [comment]
I wrote a variant of the algorithm for someone:

https://powerforum.co.za/topic/6967-adding-geyser-control-to-smart-home-wiring-question/page/2/#comment-97955

If you can use your Arduino to measure the voltage across the Geyserwise thermostat, then you can use this algorithm to determine the temperature.

**Comment by rmoosa** [comment]
Thanks. 
All of the wiring diagrams I see on the web just use one resistor. 
I noticed that u use 2 resistors and a capacitor. 
What is the purpose of the 2nd resistor and the capacitor. 
Is that cause u are using a 5v power source and not a 3.3v. Does the second resistor act as a voltage divider? And what about the capacitor. 
Thanks for ur guidance. 

Here is a sample of a diagram from the web. 
![image](https://github.com/justinschoeman/ModbusThermostat/assets/11327889/b2cf2c78-3276-4b0b-b290-86445f21d41f)

**Comment by justinschoeman** [comment]
That is the internal circuitry of the Geyserwise box. If you want to disconnect it entirely, and only use the thermostat, then you can use the circuit you pasted here.  If you want to keep the geyserwise, and just get the temperature from it, then you must just measure the voltage across the NTC and use the above algorithm.

**Comment by rmoosa** [comment]
Ah. Cool. Thanks for all the help. 
I'm planning to bypass the geyserwise totally. 
Gonna just use the gerwise thermostat and hook that up directly to a sonoff th elite. The 20A unit. 
And will probably run tasmota on the sonoff.
